File: exercices/power_fast.ml
let rec power_fast base n =
  if n = 0 then
    1.0
  else
  if n mod 2 = 1 then
    base *. power_fast (base *. base) (n / 2)
  else
    power_fast (base *. base) (n / 2)

File: exercices/power_fast.c
double power_fast(double base, int n) {
  double result = 1.0;
  int i=n;
  while (i > 0) {
    if (i % 2 == 1) {
      result = result * base;
    }
    base = base * base;
    i = i / 2;
  }
  return result;
}
